src/utils/manifold/process-cutouts.ts

```diff
@@ -0,0 +1,109 @@
+import type { ManifoldToplevel, CrossSection } from "manifold-3d/manifold.d.ts"
+import type { AnyCircuitElement, PcbCutout } from "circuit-json"
+import { su } from "@tscircuit/soup-util"
+import { SMOOTH_CIRCLE_SEGMENTS } from "../../geoms/constants"
+
+const arePointsClockwise = (points: Array<[number, number]>): boolean => {
+  let area = 0
+  for (let i = 0; i < points.length; i++) {
+    const j = (i + 1) % points.length
+    if (points[i] && points[j]) {
+      area += points[i]![0] * points[j]![1]
+      area -= points[j]![0] * points[i]![1]
+    }
+  }
+  const signedArea = area / 2
+  return signedArea <= 0
+}
+
+export interface ProcessCutoutsResult {
+  cutoutOps: any[]
+}
+
+export function processCutoutsForManifold(
+  Manifold: ManifoldToplevel["Manifold"],
+  CrossSection: ManifoldToplevel["CrossSection"],
+  circuitJson: AnyCircuitElement[],
+  pcbThickness: number,
+  manifoldInstancesForCleanup: any[],
+): ProcessCutoutsResult {
+  const cutoutOps: any[] = []
+  const pcbCutouts = su(circuitJson).pcb_cutout.list()
+
+  pcbCutouts.forEach((cutout: PcbCutout) => {
```

Comment:
ai tends to write a lot of forEach loops, but we configure biome to disable them, they are slightly more annoying to maintain for various reasons